packages/paste-core/components/menu/src/MenuBadge.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Box} from '@twilio-paste/box';
+import {Badge} from '@twilio-paste/badge';
+import type {BadgeProps} from '@twilio-paste/badge';
+import {Button} from '@twilio-paste/button';
+import type {ButtonProps} from '@twilio-paste/button';
+import {MenuPrimitiveButton} from '@twilio-paste/menu-primitive';
+import type {MenuPrimitiveButtonProps} from '@twilio-paste/menu-primitive';
+import {ChevronDownIcon} from '@twilio-paste/icons/esm/ChevronDownIcon';
+
+export type MenuBadgeProps = MenuPrimitiveButtonProps &
+  Omit<ButtonProps, 'variant'> &
+  Pick<BadgeProps, 'variant'> & {
+    buttonLabel: string;
+  };
+
+export type MenuBadgeContentProps = Omit<MenuBadgeProps, 'variant'>;
+export const MenuBadgeContent = React.forwardRef<HTMLButtonElement, MenuBadgeContentProps>(
+  ({children, element, buttonLabel, ...props}, ref) => {
+    return (
+      <Box as="span" display="flex" flexDirection="row" columnGap="space20" alignItems="center">
+        {children}
+        <MenuPrimitiveButton
+          {...props}
+          variant="secondary_icon"
+          size="reset"
+          element={`${element}_BUTTON`}
+          as={Button}
+          ref={ref}
+        >
+          <ChevronDownIcon decorative={false} size="sizeIcon10" title={buttonLabel} />
```

Comment:
question: title or `<ScreenReaderOnly>` like we do elsewhere? (ModalHeader, DataGridHeaderSort, etc)